### ISSUE TITLE
Powers of 0 over a finite field

### ISSUE THREAD

**Issue by 8d1h** [Macaulay2/M2#1856]
Hi, over a finite field, certain powers of 0 are given as 1.
```macaulay2
F = ZZ/5;
(0..10) / (i->0_F^i) -- returns (1, 0, 0, 0, 1, 0, 0, 0, 1, 0, 0)
```
If `GF` is used, the result is different.
```macaulay2
F = GF 5;
(0..10) / (i->0_F^i) -- returns (0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0)
```
Is this the expected behavior?

**Comment by mikestillman** [comment]
Hmmm, I would hope it would be consistent.  I believe that we did decide at one point that a^0 == 1, for all a. I'll look into this.  Flint is being used for both, so hopefully it will be an easy fix.

@DanGrayson (and everyone), what should our default answer be for this?

**Comment by DanGrayson** [comment]
I think it's fine if x^0 is always 1, no matter the value of x, because, at least for natural numbers, the natural way to think of m^n is as the number functions from an n element set to an m element set, so 0^0==1 is fine.

But it looks like a short is made in the results shown, based on the exponent of the multiplicative group of the field, so that x^4=1 for all x.  That's certainly not right.

**Comment by mikestillman** [comment]
@DanGrayson I have fixed this (I believe), will issue a pull request soon.

**Comment by mikestillman** [comment]
Fixed in PR #2440
